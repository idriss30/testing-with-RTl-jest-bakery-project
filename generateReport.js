// file to check how to use snapshot beyond the regular component approach
// by writting to a file
const fs = require("fs");
const os = require("os");

const path = `${os.tmpdir()}\\report.txt`;
module.exports.generateTextReport = (items) => {
  const lines = items.map(({ item, quantity, price }) => {
    return `${item} - Quantity : ${quantity} - value : ${price * quantity}`;
  });

  const total = items.reduce((sum, { price }) => {
    return sum + price;
  }, 0);

  const content = lines.concat(`Total value: ${total}`).join("\n");
  fs.writeFileSync(path, content);
};
