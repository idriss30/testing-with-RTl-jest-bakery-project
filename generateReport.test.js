import fs from "fs";
import os from "os";
import { generateTextReport } from "./generateReport.js";

const path = `${os.tmpdir()}\\report.txt`;

beforeEach(() => {});

test("generate a .txt report", () => {
  const elements = [
    { item: "cheesecake", quantity: 20, price: 5 },
    { item: "danish", quantity: 50, price: 3 },
    { item: "croissant", quantity: 500, price: 6 },
  ];

  generateTextReport(elements);
  const content = fs.readFileSync(path, "utf-8");
  expect(content).toMatchSnapshot();
});
