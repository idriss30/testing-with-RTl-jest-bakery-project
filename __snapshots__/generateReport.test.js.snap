// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate a .txt report 1`] = `
"cheesecake - Quantity : 20 - value : 100
danish - Quantity : 50 - value : 150
croissant - Quantity : 500 - value : 3000
Total value: 14"
`;
